Package python scripts

SarahAlidoost opened the pull request:
closes #71

SarahAlidoost commented:
I noticed some problems with paths in `spider_container_deploy.sh`, see issue #76

fnattino commented on src/installJDOnSLURM.py:
I have [just found out](https://packaging.python.org/en/latest/discussions/src-layout-vs-flat-layout/) the two standard layout for Python packages are the "flat layout" (package dir in the root of the repo) or "src layout" (package dir in `src/package_name/all-files-here.py`). So maybe we should move the Python scripts to a subfolder here (e.g. `src/jupyter_dask_on_slurm/..`)?

SarahAlidoost replied:
thanks, the layout is fixed.

fnattino commented on pyproject.toml:
```diff
@@ -0,0 +1,37 @@
+    "pycodestyle",
+]
+
+[tool.setuptools]
+packages = ["src"]
```
If we se the src layout (see other comment on file location), this section is not needed:  
```suggestion
```

fnattino commented on pyproject.toml:
```diff
@@ -0,0 +1,37 @@
+    "pytest",
+    "pycodestyle",
+]
+
```
We could add the CLI command in the installation, to make the script available regardless of the current working directory:

```suggestion

[project.scripts]
jupyter-dask-on-slurm = "jupyter_dask_on_slurm.runJupyterDaskOnSLURM:main"

```

fnattino commented on pyproject.toml:
```diff
@@ -0,0 +1,37 @@
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "JupyterDaskOnSLURM"
```
Let's maybe drop the capitals, that I think are discouraged for Python packages?

```suggestion
name = "jupyter-dask-on-slurm"
```

fnattino commented on src/runJupyterDaskOnSLURM.py:
```diff
@@ -44,23 +44,26 @@
 
-config_path = './config/platforms/platforms.ini'
+# get the parent pat of the current file
+parent_path = Path(__file__).resolve().parent
```
By adding one level to the dir above, I guess this would need to be updated?

SarahAlidoost replied:
I replaced it with `config_path = Path.home() / ".config/platforms/platforms.ini"` and in the user-guide.md, I asked users to create it.

fnattino commented on user-guide.md:
````diff
@@ -54,16 +54,16 @@ and, currently, [decorator](https://github.com/micheles/decorator) as well,
 
 ```shell
-python runJupyterDaskOnSLURM.py --add_platform
+python src/runJupyterDaskOnSLURM.py --add_platform
````
Also here, one more level should be added.

SarahAlidoost replied:
the command `python src/runJupyterDaskOnSLURM.py` is now replaced by `jupyter-dask-on-slurm`.

fnattino reviewed (changes requested):
Thanks a lot @SarahAlidoost , this already looks very nice! I have left a couple of comments on the `pyproject.toml`, I think we should aim to have the python package installed with a CLI to access the functionalities of the tool. This PR already brings it quite close to this goal, since it already installs dependencies automatically.

However, to make it fully working directly from the installed CLI, we need to fix a couple of things: there seems to be a circular import between the two Python files, and, most importantly, we need to figure out a global absolute path where to store the config files. I think these aspects might involve some small refactoring of the code, so I would address these in a new PR.

SarahAlidoost commented:
> Thanks a lot @SarahAlidoost , this already looks very nice! I have left a couple of comments on the `pyproject.toml`, I think we should aim to have the python package installed with a CLI to access the functionalities of the tool. This PR already brings it quite close to this goal, since it already installs dependencies automatically.
> 
> However, to make it fully working directly from the installed CLI, we need to fix a couple of things: there seems to be a circular import between the two Python files, and, most importantly, we need to figure out a global absolute path where to store the config files. I think these aspects might involve some small refactoring of the code, so I would address these in a new PR.

Thanks a lot. I addressed your comments and also fixed the circular dependencies. Here is a summary of the changes:
- the function `ssh_remote_executor` is moved to `utils.py` to fix the circular dependencies.
- the `config_path` is changed to `Path.home() / ".config/platforms/platforms.ini"` to be able to run the command` jupyter-dask-on-slurm` independent of WPD. 
- the installation command is now `pip install -U git+https://github.com/RS-DAT/JupyterDaskOnSLURM.git` which uses code in the main branch by default. No need to clone the repository anymore.
- the `wget -P` command is used in `user-guide.md` to get config files and create directories.
- The `user-guide.md` is updated accordingly.

To test the installation in this branch, do `pip install -U git+https://github.com/RS-DAT/JupyterDaskOnSLURM.git@fix_71`. I tested manually by running the commands for spider. It works fine except for `uninstall` mode due to the bug reported in issue #77 . 

I found the documentation of `Configuring dCache` is not clear. So I submitted issue #78. Please let me know if this PR  needs further improvements.
